packages/devtools/lib/src/timeline/cpu_flame_chart.dart

```diff
@@ -38,41 +36,7 @@ class CpuFlameChart extends CoreElement {
 
   bool canvasNeedsRebuild = false;
 
-  bool showingMessage = false;
-
   void _drawFlameChart() {
-    if (offlineMode &&
-        timelineController.timelineData.selectedEvent !=
-            timelineController.offlineTimelineData?.selectedEvent) {
-      final offlineModeMessage = div()
-        ..add(span(
-            text:
-                'CPU profiling is not yet available for snapshots. You can only'
-                ' view '));
-      if (timelineController.offlineTimelineData?.cpuProfileData != null) {
-        offlineModeMessage
-          ..add(span(text: 'the '))
-          ..add(span(text: 'CPU profile', c: 'message-action')
-            ..click(
-                () => timelineController.restoreCpuProfileFromOfflineData()))
-          ..add(span(text: ' included in the snapshot.'));
-      } else {
-        offlineModeMessage.add(span(
-            text:
-                'a CPU profile if it is included in the imported snapshot file.'));
-      }
-      _updateChartWithMessage(offlineModeMessage);
-      return;
-    }
-
-    if (timelineController.timelineData.cpuProfileData.stackFrames.isEmpty) {
-      _updateChartWithMessage(div(
```

Comment:
why have these error messages gone away?

Reply:
They haven't gone away - just moved up one level of abstraction to event_details.dart. This lets them be shared for the flame chart, call tree, and bottom up view.